### PR TITLE
convert Bio.Cluster to python3 only

### PR DESCRIPTION
**Pull request by mdehoon**
This pull request converts Bio.Cluster to python3.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by peterjc** [comment]
Maybe just run black on the cluster python code to solve the indentation?

File: Tests/test_Cluster.py [hunk 1]
```diff
@@ -282,6 +282,84 @@ def test_mask_parse(self):
...
+                            [1, 1, 1, 1, 1]], numpy.int32)
+        nrows, ncols = data.shape
+        clusterid = numpy.zeros(nrows, numpy.int32)
+
+        with self.assertRaisesRegex(ValueError, "^data matrix is empty$"):
+            kcluster(data[:0, :], nclusters=nclusters,
+                     mask=mask, weight=weight,
+                     transpose=False, npass=100, method="a", dist="e")
+        with self.assertRaisesRegex(ValueError,
+                "^mask has incorrect rank 1 \(expected 2\)$"):
```
Comment (peterjc) [inline Tests/test_Cluster.py]: Intend to the open bracket?

Comment (peterjc) [reply Tests/test_Cluster.py]: Or write this in the black style?

**Comment by peterjc** [comment]
Indentation fixed with minimal edit, we can do black here later.

**Comment by mdehoon** [comment]
thanks

**Pull request merged**